subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoReportAggregationPlugin.java

```diff
@@ -114,7 +119,7 @@ public void apply(Project project) {
                     executionDataConf.getIncoming().artifactView(view -> {
                         view.componentFilter(id -> id instanceof ProjectComponentIdentifier);
                         view.lenient(true);
-                    }).getFiles();
+                    }).getFiles().filter(f -> !isArchive(f)); // Filter out archive files, to work around #20532
```

Comment:
In an `ArtifactView`, you can actually look at the `ArtifactCollection` -> `ResolvedArtifactResult` -> `ResolvedVariantResult` and you could filter out `default` by name, as we should never include a legacy variant here.
What I fear with this solution is that it hampers us from adding support for bundled execution results that we would digest with an artifact transform.
Or if Jacoco's format evolves and ends up tripping that test.

Reply:
Is it a good idea to filter by variant name?  I don't think there is an expectation the names would control behavior like this, which could cause confusion.  Maybe `default` is special though and an exception could be added for it.

But I do worry that this isn't enough - there could be other configurations present other than `default` which would be accidentally selected due to this same issue (through the application of the `kotlin-jvm` plugin or another) and I think we'd want to make sure to avoid selecting jars in any case (since I think they would be the most common file format to incorrectly be selected).

I think it would be odd if Jacoco's format changed to overlap with the signature for zip files.  We could always do a version check on Jacoco and branch our logic if this happens.  If we later add support for bundled results, then wouldn't we just adjust our code to account for that later?